### PR TITLE
Add Naver sentiment movie corpus

### PR DESCRIPTION
**Pull request by jaketae**
This PR adds the [Naver sentiment movie corpus](https://github.com/e9t/nsmc), a dataset containing Korean movie reviews from Naver, the most commonly used search engine in Korea. This dataset is often used to benchmark models on Korean NLP tasks, as seen in [this paper](https://www.aclweb.org/anthology/2020.lrec-1.199.pdf).

### CODE REVIEW

File: datasets/nsmc/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  multilinguality:
+- monolingual
+  size_categories:
+- 100K<n<1M
+  source_datasets:
+- original
+  task_categories:
+- text-classification
+  task_ids:
+- sentiment-classification
```
Comment (lhoestq) [inline datasets/nsmc/README.md:19]:
```suggestion
---
annotations_creators:
- crowdsourced
language_creators:
- found
languages:
- ko
licenses:
- cc-by-1.0
multilinguality:
- monolingual
size_categories:
- 100K<n<1M
source_datasets:
- original
task_categories:
- text-classification
task_ids:
- sentiment-classification
---
```

File: datasets/nsmc/README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+# Dataset Card for [Dataset Name]
```
Comment (lhoestq) [inline datasets/nsmc/README.md:21]:
```suggestion
# Dataset Card for Naver sentiment movie corpus
```

File: datasets/nsmc/nsmc.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "document": datasets.Value("string"),
+                    "label": datasets.Value("int32"),
```
Comment (lhoestq) [inline datasets/nsmc/nsmc.py:62]: Could you use the ClassLabel feature type for the label field ?
For example
```
"label": datasets.ClassLabel(names=["negative", "positive"])
```

You'll need to regenerate the dataset_infos.json file after this change

**Review by lhoestq — COMMENTED** [review]
Looking pretty good so far ! Thanks for adding this one

I just left a comment about using the ClassLabel feature type for the label field
Could you do this change please ?

File: datasets/nsmc/nsmc.py (hunk 3 above)
Comment (jaketae) [reply datasets/nsmc/nsmc.py:62]: I was split between `ClassLabel` and integers but ended up going with the latter. On second thought, the labels make more sense. Thanks for the review!

Comment (jaketae) [reply datasets/nsmc/nsmc.py]: @lhoestq I thought I performed a proper rebase, but seems things got a little bit messy. This is probably undesirable, so I'll open a new PR. Apologies for the confusion!

**Comment by jaketae** [comment]
Closed via #1252

**Pull request closed without merging**